core/src/test/java/org/elasticsearch/fieldstats/FieldStatsTests.java

```diff
@@ -73,83 +75,153 @@ public void testLong() {
+        return new StringBuilder()
+            .append("type=").append(type)
+            .append(",index=").append(indexed)
+            .append(",doc_values=").append(docValues)
+            .append(",store=").append(stored).toString();
+    }
+
     public void testString() {
-        createIndex("test", Settings.EMPTY, "test", "field", "type=text");
+        createIndex("test", Settings.EMPTY, "test",
+            "field_index", makeType("keyword", true, false, false),
+            "field_dv", makeType("keyword", false, true, false),
+            "field_stored", makeType("keyword", false, true, true),
+            "field_source", makeType("keyword", false, false, false));
         for (int value = 0; value <= 10; value++) {
-            client().prepareIndex("test", "test").setSource("field",
-                String.format(Locale.ENGLISH, "%03d", value)).get();
+            String keyword =  String.format(Locale.ENGLISH, "%03d", value);
+            client().prepareIndex("test", "test")
+                .setSource("field_index", keyword,
+                    "field_dv", keyword,
+                    "field_stored", keyword,
+                    "field_source", keyword).get();
         }
         client().admin().indices().prepareRefresh().get();
 
-        FieldStatsResponse result = client().prepareFieldStats().setFields("field").get();
-        assertThat(result.getAllFieldStats().get("field").getMaxDoc(), equalTo(11L));
-        assertThat(result.getAllFieldStats().get("field").getDocCount(), equalTo(11L));
-        assertThat(result.getAllFieldStats().get("field").getDensity(), equalTo(100));
-        assertThat(result.getAllFieldStats().get("field").getMinValue(),
-            equalTo(new BytesRef(String.format(Locale.ENGLISH, "%03d", 0))));
-        assertThat(result.getAllFieldStats().get("field").getMaxValue(),
-            equalTo(new BytesRef(String.format(Locale.ENGLISH, "%03d", 10))));
-        assertThat(result.getAllFieldStats().get("field").getMinValueAsString(),
-            equalTo(String.format(Locale.ENGLISH, "%03d", 0)));
-        assertThat(result.getAllFieldStats().get("field").getMaxValueAsString(),
-            equalTo(String.format(Locale.ENGLISH, "%03d", 10)));
-        assertThat(result.getAllFieldStats().get("field").getDisplayType(),
-            equalTo("string"));
+        FieldStatsResponse result = client().prepareFieldStats()
+            .setFields("field_index", "field_dv", "field_stored", "field_source").get();
+        assertThat(result.getAllFieldStats().size(), equalTo(3));
+        for (String field : new String[] {"field_index", "field_dv", "field_stored"}) {
+            assertThat(result.getAllFieldStats().get(field).getMaxDoc(), equalTo(11L));
+            assertThat(result.getAllFieldStats().get(field).getDisplayType(),
+                equalTo("string"));
+            if ("field_index".equals(field)) {
+                assertThat(result.getAllFieldStats().get(field).getMinValue(),
```

Comment:
I'm not really a fan of `assertThat(..., equalTo(...))`. I feel like it is usually more clear to do `assertEquals(...)`. Like, if you have lots of assertions and you think they should line up then fine, but in this case I think they are all `equalTo` so maybe switch? Not a big deal either way.